featuretools/computational_backends/calculate_feature_matrix.py

```diff
@@ -172,61 +172,25 @@ def calculate_feature_matrix(features, entityset=None, cutoff_time=None, instanc
-            raise AttributeError('Cutoff time DataFrame must contain a column with either the same name'
-                                 ' as the target entity index or a column named "instance_id"')
-        # rename to instance_id
-        cutoff_time.rename(columns={target_entity.index: "instance_id"}, inplace=True)
-
-    if "time" not in cutoff_time.columns:
-        if target_entity.time_index and target_entity.time_index not in cutoff_time.columns:
-            raise AttributeError('Cutoff time DataFrame must contain a column with either the same name'
-                                 ' as the target entity time_index or a column named "time"')
-        # rename to time
-        cutoff_time.rename(columns={target_entity.time_index: "time"}, inplace=True)
-
-    # Make sure user supplies only one valid name for instance id and time columns
-    if "instance_id" in cutoff_time.columns and target_entity.index in cutoff_time.columns and \
-            "instance_id" != target_entity.index:
-        raise AttributeError('Cutoff time DataFrame cannot contain both a column named "instance_id" and a column'
-                             ' with the same name as the target entity index')
-    if "time" in cutoff_time.columns and target_entity.time_index in cutoff_time.columns and \
-            "time" != target_entity.time_index:
-        raise AttributeError('Cutoff time DataFrame cannot contain both a column named "time" and a column'
-                             ' with the same name as the target entity time index')
-
-    # Check that cutoff_time time type matches entityset time type
-    if entityset.time_type == NumericTimeIndex:
-        if cutoff_time['time'].dtype.name not in PandasTypes._pandas_numerics:
-            raise TypeError("cutoff_time times must be numeric: try casting "
-                            "via pd.to_numeric(cutoff_time['time'])")
-    elif entityset.time_type == DatetimeTimeIndex:
-        if cutoff_time['time'].dtype.name not in PandasTypes._pandas_datetimes:
-            raise TypeError(
-                "cutoff_time times must be datetime type: try casting via pd.to_datetime(cutoff_time['time'])")
-    assert (cutoff_time[['instance_id', 'time']].duplicated().sum() == 0), \
-        "Duplicated rows in cutoff time dataframe."
-
-    pass_columns = [col for col in cutoff_time.columns if col not in ['instance_id', 'time']]
-
-    if _check_time_type(cutoff_time['time'].iloc[0]) is None:
-        raise ValueError("cutoff_time time values must be datetime or numeric")
-
-    # make sure dtype of instance_id in cutoff time
-    # is same as column it references
-    target_entity = features[0].entity
-    dtype = entityset[target_entity.id].df[target_entity.index].dtype
-    cutoff_time["instance_id"] = cutoff_time["instance_id"].astype(dtype)
+            instance_ids = df[index_var]
+
+        if isinstance(instance_ids, dd.Series):
+            instance_ids = instance_ids.compute()
```

Comment:
Warning here could be confusing. This `.compute()` is needed in the case where the target entity is a Dask dataframe, but `instance_ids` was `None`. In that case, the instance ids are created from the dataframe index column, but we need to compute this because the length of `instance_ids` is used by `_handle_chunk_size` and for the progress bar. Was thinking it would be better to compute once here rather than computing multiple times whenever the length is needed.

I think this could be cleaned up in the future with some additional refactoring.